packages/@aws-cdk/aws-apigateway/lib/restapi.ts

```diff
@@ -427,18 +343,190 @@ export class RestApi extends Resource implements IRestApi {
       }
     }
   }
+}
 
-  private configureCloudWatchRole(apiResource: CfnRestApi) {
-    const role = new iam.Role(this, 'CloudWatchRole', {
-      assumedBy: new iam.ServicePrincipal('apigateway.amazonaws.com'),
-      managedPolicies: [iam.ManagedPolicy.fromAwsManagedPolicyName('service-role/AmazonAPIGatewayPushToCloudWatchLogs')],
+/**
+ * Represents a REST API in Amazon API Gateway, created via a Swagger/OpenAPI specification.
+ *
+ * Some properties normally accessible on @see {@link RestApi} - such as the description -
+ * must be declared in the specification.
+ *
+ * By default, the API will automatically be deployed and accessible from a
+ * public endpoint.
+ *
+ * @resource AWS::ApiGateway::RestApi
+ */
+export class APIDefinitionRestApi extends BaseRestApi {
```

Comment:
_(Randomly jumping in. I've followed the thread and it's been very informational.)_

From where did `ApiDefinition` originate? I see [this API Gateway doc](https://docs.aws.amazon.com/apigateway/latest/developerguide/api-gateway-import-api.html) uses a similar phrase, so I'm assuming "from APIG." (emphasis mine)

> Currently, API Gateway supports OpenAPI v2.0 and OpenAPI v3.0 **definition files**

Therefore, while Api...Api is a bit odd of a name, it seems to answer the question, "What type of REST API?" pretty well: "An API Definition REST API" (rather than lambda or plain old rest API).

Also, I'm assuming these constructs work with both Swagger (Open API 2) and Open API 3, so this is not `OpenApi3RestApi` or similar. (_Why I bring up this point_) If [Smithy](https://github.com/awslabs/smithy) was added to APIG in the future (directly, not via an Open API translation), do we think it would be `ApiDefinitionRestApi` as well or would it be separate (`SmithyRestApi`)?

Reply:
I wouldn't speculate this too far - they have used the same (CloudFormation) property to support both OpenAPIv2 and OpenAPIv3, so I would suspect they can use the same mechanism to also do Smithy.
Further, with the new APIGWv2 launched, it's not clear how much features they'll add to the first gen RestApi.

@bdgmb2 - how about `SpecRestApi`?

Reply:
I am fine with either `SpecRestApi` and `ApiDefinitionRestApi` as a consumer. To your point, `Spec` may stop future people from groaning at `Api...Api`, so that sounds good.